### PR TITLE
[Version Upgrade] Remove bootstrap and react-bootstrap and reimplement using react-modal

### PR DESCRIPTION
**Pull request by robsli**
## Description

As part of the Core Web Vitals work, we removed bootstrap from Stockblocks in favor of using tailwindcss for styling. It seems that removing the core bootstrap javascript that was on Stockblocks broke the "X" button on the NPSModal. 

It seems that `NPSModal` uses older versions of `bootstrap` ([3.3.7](https://www.npmjs.com/package/bootstrap/v/3.3.7)) and `react-bootstrap` ([0.33.1](https://www.npmjs.com/package/react-bootstrap/v/0.33.1)), so this was an opportunity to also update dependencies.

I opted to replace bootstrap with [`react-modal`](http://reactcommunity.org/react-modal/) in an effort to move away from requiring bootstrap in our dependencies and to move the behavior closer to how the [`ModalDialog`](https://github.com/Footage-Firm/StockBlocks/blob/master/assets/react/common/components/ModalDialog.js) component works in Stockblocks.

I also refreshed the styles to follow more closely to the [`Modal Dialog` guidance in the design system documentation](https://www.notion.so/storyblocks/Modal-Dialog-62c7d92a4d584be6a86ebdc778bf80e3).

Sorry in advance if I offended anyone by reaching for plain CSS instead of using `styl`! I haven't used it before and couldn't adjust easily, so decided to go with something more familiar 😅 I'm happy to go through and use re-style using `styl` syntax if there's a strong desire to keep this component styled that way.

### Summary of Changes

1. Removed `bootstrap` and `react-bootstrap` dependencies and pulled in `react-modal`
2. Added the `overlayClassName` property to allow for custom styling of the overlay
3. Refactored `NPSModal.js` using react hooks
4. Updated `README`

#### Default

nps-modal-default-style

#### Custom (using `revshare` class)

nps-modal-custom-styling

### CODE REVIEW

**Review by jasonvideoblocks — APPROVED** [review]
LGTM!  👍  long as any other consumers of the package are good with it

File: stylesheets/nps-modal.css [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+    --color-gray-700: #626C78;
+    --color-gray-900: #1f2830;
+    --shadow-popover: 0 1.5rem 3rem 0 rgba(0, 0, 0, 0.15), 0 0.5rem 0.75rem 0 rgba(0, 0, 0, 0.20);
+
+    display: flex;
+    flex-flow: column nowrap;
+    font-size: 16px;
+}
+
+.nps-modal__overlay {
```
Comment (JennStoryblocks) [inline stylesheets/nps-modal.css:14]: What's the reasoning for the `__` here?

File: stylesheets/nps-modal.css [hunk 2]
```diff
@@ -0,0 +1,155 @@
+/* Modal Styles */
```
Comment (JennStoryblocks) [inline stylesheets/nps-modal.css:1]: Was there a reason for moving from `styl` -> `css` here?

**Review by JennStoryblocks — APPROVED** [review]
LGTM! Just a couple qs

File: stylesheets/nps-modal.css (hunk 1 above)
Comment (robsli) [reply stylesheets/nps-modal.css:14]: It follows the [BEM class naming guide](http://getbem.com/naming/)

File: stylesheets/nps-modal.css (hunk 2 above)
Comment (robsli) [reply stylesheets/nps-modal.css:1]: Not particularly, I just hadn't used `styl` before so didn't want to run into any weird quirks. I can move it back to `styl` if you'd prefer to keep it like that though!

**Review by vikramp93 — APPROVED** [review]
LGTM

File: stylesheets/nps-modal.css (hunk 2 above)
Comment (JennStoryblocks) [reply stylesheets/nps-modal.css:1]: You should be good with this! Ideally we move it to tailwind eventually anywayssss

Comment (robsli) [reply stylesheets/nps-modal.css:1]: Sweet! I definitely tried to use tailwind before realizing the classes weren't doing anything 😅

**Pull request merged**